src/payment.coffee

```diff
@@ -131,44 +127,81 @@ hasTextSelected = (target) ->
 
 reFormatCardNumber = (e) ->
   setTimeout =>
-    target = e.target
+    target  = e.target
     value   = QJ.val(target)
-    value   = Payment.fns.formatCardNumber(value)
-    QJ.val(target, value)
+
+    # clear the value
+    QJ.val target, ''
```

Comment:
Why was this needed?

Currently, when you paste something in the inputs of the card, none of the filters are being triggered and you can paste whatever you want. Every check is being ignored.

By simulating typing, you trigger the formatters (by emitting Events), one character at a time and whatever you paste is being validated and formatted.

Its not elegant and its hack-ish, i agree. Problem is that almost none of the browsers offer a clean way to do this and this works so far in the majority of the browsers.